core/packages/contracts/contracts/utils/Bitfield.sol

```diff
@@ -48,87 +41,103 @@ library Bitfield {
         bitfield = new uint256[](prior.length);
         uint256 found = 0;
 
-        for (uint256 i = 0; found < n; i++) {
+        for (uint256 i = 0; found < n;) {
             bytes32 randomness = keccak256(abi.encodePacked(seed, i));
-            uint256 index = uint256(randomness) % length;
+            uint256 index;
+            unchecked {
+                index = uint256(randomness) % length;
+            }
+
+            Location memory loc = toLocation(index);
 
             // require randomly selected bit to be set in prior
-            if (!isSet(prior, index)) {
+            if (!isSet(prior, loc)) {
+                unchecked { ++i; }
                 continue;
             }
 
             // require a not yet set (new) bit to be set
-            if (isSet(bitfield, index)) {
+            if (isSet(bitfield, loc)) {
+                unchecked { ++i; }
                 continue;
```

Comment:
I have a feeling we can combine these two checks into a single operation with some fancy bit twiddling. 

```solidity
if (IsNotSetInAandIsSetInB(a, b, loc)) {
  unchecked { i++; }
  continue;
}
```